OpenBSD Fails to build D++ due to broken <locale> system header

i cant compile DPP

i get this error

```
In file included from /home/mrtuxa/pkgs/DPP/src/dpp/discordevents.cpp:25:
In file included from /usr/include/c++/v1/iostream:38:
In file included from /usr/include/c++/v1/istream:163:
In file included from /usr/include/c++/v1/ostream:140:
In file included from /usr/include/c++/v1/locale:218:
/usr/include/c++/v1/__bsd_locale_fallbacks.h:122:17: error: use of undeclared identifier 'vasprintf'; did you mean 'vsprintf'?
    int __res = vasprintf(__s, __format, __va);
                ^
/usr/include/c++/v1/cstdio:124:9: note: 'vsprintf' declared here
using ::vsprintf _LIBCPP_USING_IF_EXISTS;
        ^
In file included from /home/mrtuxa/pkgs/DPP/src/dpp/discordevents.cpp:25:
In file included from /usr/include/c++/v1/iostream:38:
In file included from /usr/include/c++/v1/istream:163:
In file included from /usr/include/c++/v1/ostream:140:
In file included from /usr/include/c++/v1/locale:218:
/usr/include/c++/v1/__bsd_locale_fallbacks.h:122:27: error: cannot initialize a parameter of type 'char *' with an lvalue of type 'char **'
    int __res = vasprintf(__s, __format, __va);
                          ^~~
/usr/include/stdio.h:269:21: note: passing argument to parameter here
int      vsprintf(char *, const char *, __va_list);
                        ^
[ 71%] Building CXX object library/CMakeFiles/dpp.dir/__/src/dpp/discordclient.cpp.o
2 errors generated.
*** Error 1 in target 'library/CMakeFiles/dpp.dir/__/src/dpp/discordevents.cpp.o'
*** Error 1 in . (library/CMakeFiles/dpp.dir/build.make:538 'library/CMakeFiles/dpp.dir/__/src/dpp/discordevents.cpp.o': cd /home/mrtuxa/pkg...)
*** Error 2 in . (CMakeFiles/Makefile2:102 'library/CMakeFiles/dpp.dir/all')
*** Error 2 in /home/mrtuxa/pkgs/DPP/build (Makefile:156 'all': /usr/bin/make -s -f CMakeFiles/Makefile2 all)
```

--- a/src/dpp/discordevents.cpp
+++ b/src/dpp/discordevents.cpp
@@ -18,9 +18,16 @@
  * limitations under the License.
  *
  ************************************************************************************/
-#ifndef _XOPEN_SOURCE
-	#define _XOPEN_SOURCE
+
+/* OpenBSD errors when xopen_source is defined.
+ * We want to make sure that OpenBSD does not define it.
+ */
+#if !defined(__OpenBSD__)
+	#ifndef _XOPEN_SOURCE
+		#define _XOPEN_SOURCE
+	#endif
 #endif
+
 #include <string>
 #include <iostream>
 #include <fstream>

